### PR TITLE
PNG: fix interlaced images that have a width or height that is not a multiple of 8

### PR DESCRIPTION
**Pull request by melissalinkert**
Fixes https://trac.openmicroscopy.org/ome/ticket/12850.

See also
http://www.libpng.org/pub/png/spec/1.2/PNG-DataRep.html#DR.Interlaced-data-order.

To test, use the file from QA 11038.  Without this change, ```showinf``` on the file should throw an exception; with this change, ```showinf``` should produce the same image as ImageMagick (the ```display``` command) and ```File > Open``` in ImageJ.

### CODE REVIEW

**Comment by melissalinkert** [comment]
Fix pushed for https://ci.openmicroscopy.org/view/Bio-Formats/job/BIOFORMATS-5.1-merge-full-repository/535/

File: components/formats-bsd/src/loci/formats/in/APNGReader.java [hunk 1]
```diff
@@ -420,12 +437,168 @@ else if (compression == 0) {
...
+                break;
+              case 5:
+                if (i == 3) {
+                  passWidth -= (PASS_WIDTHS[i] - 1);
+                }
+                if (i == 5) {
+                  passWidth -= (PASS_WIDTHS[i] - 2);
+                }
+                if (i == 4) {
+                  passWidth -= (PASS_WIDTHS[i] =- 3);
```
Comment (mtbc) [inline components/formats-bsd/src/loci/formats/in/APNGReader.java]: is this second `=` desired? (also could make these "else if"'s but doesn't much matter really)

**Comment by mtbc** [comment]
It's a strange looking image but other viewers agree that's what it looks like! So, this PR looks good functionally. :+1:

File: components/formats-bsd/src/loci/formats/in/APNGReader.java (hunk 1 above)
Comment (melissalinkert) [inline components/formats-bsd/src/loci/formats/in/APNGReader.java]: No, that's just a typo - fix pushed.

**Pull request merged**